third_party/amd/lib/TritonAMDGPUToLLVM/MemoryOpToLLVM.cpp

```diff
@@ -215,10 +215,12 @@ class LocalBarrierOpConversion
     // - s_waitcnt specifies how many operations to VMEM/LDS can be outstanding
     //   when the instruction completes.
     //   In this case we require 0 outstanding LDS operations
+    //   amdgpu::MemoryCounterWaitOp will lower s_waitcnt
     // - s_barrier syncronizes the execution for the CTA
-    constexpr int32_t ldsOnlyBits = ~(0x1f << 8);
-    Location loc = op->getLoc();
-    ROCDL::SWaitcntOp::create(rewriter, loc, ldsOnlyBits);
+    auto dsAttr = rewriter.getI32IntegerAttr(0);
+    rewriter.create<amdgpu::MemoryCounterWaitOp>(
+        op->getLoc(), /* load= */ nullptr, /* store= */ nullptr,
```

Comment:
Nit: Interestingly clang-format accepts this? I'd expect it wants `/*load=*/` without spaces inside.